thunder/executors/sdpaex.py

```diff
@@ -301,10 +301,12 @@ def _scaled_dot_product_efficient_attention_backward_impl(
     if attn_mask is None:
         grad_input_mask.append(False)
     else:
-        grad_input_mask.append(attn_mask.requires_grad)
+        # Cannot rely on the requires_grad in the meta function,
```

Comment:
`requires_grad` of intermediate TensorProxies is ignored in our automatic differentiation code because we haven't done the work of properly threading this property through all computations.
We should remove the ability to query `.requires_grad` from intermediate TensorProxies completely to avoid similar bugs in the future. This can be achieved by introducing a separate "InputTensorProxy" which has this attribute and removing it from the regular TensorProxy.